src/main/java/edu/harvard/iq/dataverse/marketplace/controller/api/AuthController.java

```diff
@@ -0,0 +1,240 @@
+ */
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    UserRepo userRepository;
+
+    @Autowired
+    RoleRepo roleRepository;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Autowired
+    JwtUtils jwtUtils;
+
+    @PostMapping("/login")
+    @AuthAPIDocs.Login
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
+
+        Authentication authentication = authenticationManager
+                .authenticate(new UsernamePasswordAuthenticationToken(loginRequest.getUsername(),
+                        loginRequest.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        
+        String jwt = jwtUtils.generateJwtToken(authentication);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+        List<String> roles = userDetails.getAuthorities().stream().map(item -> item.getAuthority())
+                .collect(Collectors.toList());
+
+        return ResponseEntity
+                .ok(new JwtResponse(jwt,
+                        userDetails.getId(),
+                        userDetails.getUsername(),
+                        userDetails.getEmail(),
+                        roles));
+    }
+
+    
+    @PreAuthorize(ApplicationRoles.ADMIN_ROLE)
+    @PostMapping("/signup")
+    @AuthAPIDocs.Signup
+    public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRequest) {
+
+        ServerMessageResponse messageResponse = null;
+
+        if (userRepository.existsByUsername(signUpRequest.getUsername())) {
+            messageResponse = new ServerMessageResponse(HttpStatus.BAD_REQUEST,
+                    "Username is already taken.",
+                    "Error during the creation of the user, a user with this username already exist.");
+
+            return ResponseEntity.badRequest().body(messageResponse);
+        }
+
+        if (userRepository.existsByEmail(signUpRequest.getEmail())) {
+            messageResponse = new ServerMessageResponse(HttpStatus.BAD_REQUEST,
+                    "Email is already taken.",
+                    "Error during the creation of the user, a user with this email already exist.");
+            return ResponseEntity.badRequest().body(messageResponse);
+        }
+
+        User user = new User(signUpRequest.getUsername(),
+                signUpRequest.getEmail(),
+                encoder.encode(signUpRequest.getPassword()));
+
+        userRepository.save(user);
+
+        messageResponse = new ServerMessageResponse(HttpStatus.OK,
+                "User registered successfully.",
+                "The user was successfully registered.");
+
+        return ResponseEntity.ok(messageResponse);
+    }
+
+    
+    @PreAuthorize(ApplicationRoles.ADMIN_ROLE)
+    @PostMapping("/role")
```

Comment:
We already have "/api/auth/roles" below. Should we use "roles" instead of "role" here.

And also changes like this:

```
-    @PostMapping("/role/{roleId}/user/{userId}")
+     @PostMapping("/roles/{roleId}/user/{userId}")
```

etc. etc. What do you think?